feat(react-native-webrtc): adds config mod for Android minSdkVersion 24

thorbenprimke opened the pull request:
<!--
🚨 We use semantic release (a bot publishes everything automatically)
🚨 so please use conventional commit style for the PR title:
🚨 https://www.conventionalcommits.org/en/v1.0.0/
🚨 Example: feat(detox): add new step
-->

# Why

react-native-webrtc requires that the Android minSdkVersion is
set to 24.
Related issue on this [here](https://github.com/react-native-webrtc/react-native-webrtc/issues/720) and also mentioned on [Daily's RN SDK](https://github.com/daily-co/react-native-daily-js)
that has react-native-webrtc as a dependency.

Adds withBuildScriptExtMinimumVersion same as the ffmpeg-kit-react-native
config plugin. Maybe withBuildScriptExtMinimumVersion could be
exported in `config-plugins` package so that it could be used directly.

# How

Added withBuildScriptExtMinimumVersion plugin and used it to set the minSdkVersion to 24

# Test Plan

1) Created a patch-package with the code changes.
2) Verified with `expo prebuild -p android` that the generated source now have minSdkVersion set to 24. By default / without the config plugin change it would be 21.

EvanBacon commented on packages/react-native-webrtc/src/withBuildScriptExtMinimumVersion.ts:
```diff
@@ -0,0 +1,50 @@
+} from "@expo/config-plugins";
+
+/** Sets a numeric version for a value in the project.gradle buildscript.ext object to be at least the provided props.minVersion, if the existing value is greater then no change will be made. */
+export const withBuildScriptExtMinimumVersion: ConfigPlugin<{
```
I think you can use `import { AndroidConfig } from '@expo/config-plugins'` then `AndroidConfig.Version.withBuildScriptExtMinimumVersion`

EvanBacon replied:
Refer to this more updated plugin https://github.com/expo/config-plugins/blob/379edcd74cc454faf6ca108ea60faffbf7b3a9c3/packages/react-native-adjust/src/withReactNativeAdjust.ts#L97-L100

EvanBacon reviewed (changes requested):
Could you also add a note to the README which informs users about the minimum version change, it's one of the bigger changes a plugin could make.

thorbenprimke commented:
> Could you also add a note to the README which informs users about the minimum version change, it's one of the bigger changes a plugin could make.

Thanks for the review @EvanBacon. I updated the PR and added the note to the README.
